app/views/books/_form.html.erb

```diff
@@ -17,10 +17,48 @@
   </div>
 
   <div class="field">
-    <%= form.label :publisher_id %>
-    <%= form.text_field :publisher_id %>
+    <%= label :publisher, :name, "Publisher"%>
+    <% if book.publisher.present? %>
+      <%= text_field :publisher, :name, value: book.publisher.name %>
+    <% else %>
+      <%= text_field :publisher, :name %>
+    <% end %>
   </div>
 
+  <div id="authors_field" class="field">
+    <template id="author_field_template">
+      <div class="removable_field">
+        <input type="text" name="authors[][name]" id="author_name"/>
+        <input type="button" value="X" class="delete_item" />
+      </div>
+    </template>
+    <%= label :authors, :name, "Authors"%>
+      <% book.authors.each do |value| %>
+        <div class="removable_field">
+          <%= text_field "author", "name", value:"#{value[:name]}", name: "authors[][name]" %>
+          <input type="button" value="X" class="delete_item" />
+        </div>
+      <% end %>
+  </div>
+  <input type="button" value="New Author" id="add_author" />
+
+  <div id="categories_field" class="field">
+    <template id="category_field_template">
+      <div class="removable_field">
+        <input type="text" name="categories[][name]" id="category_name"/>
+        <input type="button" value="X" class="delete_item" />
+      </div>
+    </template>
+    <%= label :categories, :name, "Categories"%>
+      <% book.categories.each do |value| %>
+        <div class="removable_field">
+          <%= text_field "category", "name", value:"#{value[:name]}", name: "categories[][name]" %>
+          <input type="button" value="X" class="delete_item" />
+        </div>
+      <% end %>
+  </div>
+  <input type="button" value="New Category" id="add_category" />
```

Comment:
general comment: why are you using html tags like `input` instead of erb tags?

Reply:
What is the difference?

Reply:
It is very hard to read mixed erb and html tags.
Also make use of partials

Reply:
ok